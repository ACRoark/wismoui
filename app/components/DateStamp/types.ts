export type DateStampFormat = 'auto' | 'long' | 'short';
