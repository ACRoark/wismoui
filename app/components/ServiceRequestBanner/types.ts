export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
