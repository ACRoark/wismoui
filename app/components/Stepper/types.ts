import { ReactElement } from 'react';
import { Description } from 'types';

export interface IStepDescription {
  description: Description;
  key: string;
  status: StepperStatus;
  title: ReactElement | string;
}

export enum StepperStatus {
  Active = 'active',
  Completed = 'completed',
  Error = 'error',
  Pending = 'pending',
}
