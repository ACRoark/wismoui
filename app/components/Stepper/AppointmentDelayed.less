@import (reference) '../../styles/themes/index.less';

@appointment-delayed-cls-prefix: ~'@{dte-prefix}-appointment-delayed';

.@{appointment-delayed-cls-prefix} {
  .message {
    color: @dte-red;
    font-weight: @dte-wismo-font-weight-semi-bold;
  }

  .view-details-link {
    font-size: @dte-wismo-font-size-0;
    font-weight: @dte-wismo-font-weight-semi-bold;
    padding: unset;
    padding-left: 13px;
  }
}
