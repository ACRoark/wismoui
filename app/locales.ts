export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
