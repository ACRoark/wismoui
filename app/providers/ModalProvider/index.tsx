/**
 *
 * ModalProvider
 *
 */
import React, { useState } from 'react';

import IModalConfig from 'types/IModalConfig';
import IModalState from 'types/IModalState';

import Modal from './Modal';

import 'react-responsive-modal/styles.css';

interface IModalProviderProps {
  children?: React.ReactNode;
}

const ModalContext = React.createContext<IModalState | null>(null);

const ModalConsumer = ModalContext.Consumer;

const ModalProvider: React.FC<IModalProviderProps> = (props: IModalProviderProps): React.ReactElement => {
  const { children } = props;

  const [modalConfig, setModalConfig] = useState<IModalConfig | null>(null);

  const showModal = (config: IModalConfig): void => {
    setModalConfig(config);
  };

  const hideModal = (): void => {
    setModalConfig(null);
  };

  return (
    <ModalContext.Provider
      value={{
        hideModal,
        showModal,
      }}
    >
      {React.Children.only(children)}
      {modalConfig && <Modal onClose={hideModal} config={modalConfig} />}
    </ModalContext.Provider>
  );
};

export { ModalConsumer, ModalContext, ModalProvider };
