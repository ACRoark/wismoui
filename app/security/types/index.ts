import IAuthenticationState from 'types/IAuthenticationState';

import IAuthenticationResult from './IAuthenticationResult';

export { IAuthenticationResult };

export default IAuthenticationState;
