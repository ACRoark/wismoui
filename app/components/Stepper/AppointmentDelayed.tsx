import React from 'react';
import { FormattedMessage } from 'react-intl';

import DateStamp from 'components/DateStamp';
import getServiceOrderEventsModal2 from 'components/ViewDetailsLink/getServiceOrderEventsModal2';
import ViewSubstatusDetailsLink from 'components/ViewDetailsLink/ViewSubstatusDetailsLink';
import useConfig from 'hooks/useConfig';
import { ProductType } from 'types';

import getAppointmentType from './getAppointmentType';
import messages from './messages';

import './AppointmentDelayed.less';

interface IAppointmentDelayedProps {
  date: string;
  product: ProductType;
}

const AppointmentDelayed: React.FC<IAppointmentDelayedProps> = (
  props: IAppointmentDelayedProps,
): React.ReactElement => {
  const { date, product } = props;
  const { urls } = useConfig();

  const serviceType = getAppointmentType(product);

  const modal = getServiceOrderEventsModal2('DELAYED', urls.backToMIMO);

  return (
    <div className="dte-wismo-appointment-delayed">
      <span className="message">
        <FormattedMessage
          {...messages.DELAYED}
          values={{
            date: <DateStamp format="long" value={date} />,
          }}
        />
        <ViewSubstatusDetailsLink className="view-details-link" modal={modal} />
      </span>
      <div>
        <FormattedMessage {...messages[serviceType]} values={{ window: null }} />
      </div>
    </div>
  );
};

export default AppointmentDelayed;
