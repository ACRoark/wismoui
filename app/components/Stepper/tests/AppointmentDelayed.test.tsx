import React from 'react';
import { Provider } from 'react-redux';

import { SUPPORTED_LOCALES } from 'locales';

import { ModalProvider } from 'providers/ModalProvider';
import { createMockStore, createSnapshotWithIntl } from 'testing/utils';
import { IServiceOrderEvent, ProductType } from 'types';

import AppointmentDelayed from '../AppointmentDelayed';

const expectComponentToMatchSnapshot = (
  date: string,
  product: ProductType,
): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);
  languages.forEach((language: string): void => {
    const tree = createSnapshotWithIntl(
      <Provider store={createMockStore()}>
        <ModalProvider>
          <AppointmentDelayed product={product} date={date} />
        </ModalProvider>
      </Provider>, language
    );
    expect(tree).toMatchSnapshot();
  });
};

describe('<AppointmentDelayed />', (): void => {

  const serviceOrderEvents: IServiceOrderEvent[] = [
    {
      appointment: {
        date: '20200622',
        slotType: 'AM',
      },
      latestStatus: {
        createdAtDate: '20200622',
        createdAtTime: '223443',
        serviceOrderEventStatus: 'DELAYED',
      },
      product: 'EFR_D1',
      serviceOrderCategory: null,
      serviceOrderId: 'abc123',
    },
  ];

  it('should render correctly when the required props are given', (): void => {
    const {appointment: {date}, product} = serviceOrderEvents[0];

    expectComponentToMatchSnapshot( date, product);
  });
});
