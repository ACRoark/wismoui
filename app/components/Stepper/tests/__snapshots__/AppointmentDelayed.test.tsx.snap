// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppointmentDelayed /> should render correctly when the required props are given 1`] = `
<div
  className="dte-wismo-appointment-delayed"
>
  <span
    className="message"
  >
    Monday
    , 
    June
     
    22
     - Order Delayed
    <button
      className="view-details-link dte-wismo-button dte-wismo-button-link"
      data-track="mimo"
      data-track-action="click"
      data-track-detail="order tracker"
      onClick={[Function]}
    >
      View Details
    </button>
  </span>
  <div>
    Electric appointment 
  </div>
</div>
`;

exports[`<AppointmentDelayed /> should render correctly when the required props are given 2`] = `
<div
  className="dte-wismo-appointment-delayed"
>
  <span
    className="message"
  >
    Lunes
    , 
    Junio
     
    22
     - Pedido retrasado
    <button
      className="view-details-link dte-wismo-button dte-wismo-button-link"
      data-track="mimo"
      data-track-action="click"
      data-track-detail="order tracker"
      onClick={[Function]}
    >
      Ver Detalles
    </button>
  </span>
  <div>
    Cita electrica 
  </div>
</div>
`;
